### PR TITLE
Added Week 1 Assessment files

### PR DESCRIPTION
**Pull request by mikerogers04**
# Submitting Week 1 Assessment

### ✅ Completed Files

Select the following as confirmation that the files in this repository are complete.

- [x] Coding Challenges
- [x] Coding Concepts
- [x] Interview Questions

### 📝 Feedback

If you would like specific feedback on a particular problem please leave details here:
Please provide feedback on Interview questions/answers.

### CODE REVIEW

File: code-challenges.js [hunk 1]
```diff
@@ -20,49 +20,85 @@ const fruit4 = "kiwi"
...
 const temp2 = 350
 // Expected output: "350 is above boiling point"
-
+console.log(boilingPoint(temp2))  //console log after set paramaters, cannot be before initialization
 const temp3 = 212
 // Expected output: "212 is at boiling point"
+console.log(boilingPoint(temp3))  //console log after set paramaters, cannot be before initialization
+const error = "hi" 
+// Exepected output: Error: Please input a number value"
+console.log(boilingPoint(error)) //Extra console log to verify the correct parameter is being passed. In this case a string is being passed, when the function expects a number, therefore an error message will output
```
Comment (TsunamiCode5) [inline code-challenges.js:50]: This is amazing!

File: code-challenges.js [hunk 2]
```diff
@@ -20,49 +20,85 @@ const fruit4 = "kiwi"
 
 // --------------------1) Create a function that takes in a number and determines if the given number is below boiling point, at boiling point, or above boiling point. Boiling point is 212 degrees Fahrenheit. Use the test variables provided below.
 
-// Pseudo code:
-
+// Pseudo code: create a function called boilingPoint that passes in a parameter of a number.
```
Comment (TsunamiCode5) [inline code-challenges.js:23]: Beautiful pseudocode! 🙌🏽

File: code-challenges.js [hunk 3]
```diff
@@ -20,49 +20,85 @@ const fruit4 = "kiwi"
...
 // --------------------2) Create the code that will combine the two arrays and return the length using the test variables provided below.
 
-// Pseudo code:
+// Pseudo code: concat both arrays into one a newly created array called combinedArray. Use .length method to return the length of the combined array
 
 const padres1984WorldSeriesRuns = [2, 5, 2, 2, 4]
 const padres1998WorldSeriesRuns = [6, 3, 5, 3]
 // Expected output: 9
 
+var combinedArray = padres1984WorldSeriesRuns.concat(padres1998WorldSeriesRuns)
```
Comment (TsunamiCode5) [inline code-challenges.js:60]: Nice!

File: code-challenges.js [hunk 4]
```diff
@@ -20,49 +20,85 @@ const fruit4 = "kiwi"
...
+// Pseudo code: use .sort methoed on the both arrays. Include optional argument using compare function to sort the numbers from highest to lowest. .sort method sorts values as strings, both arrays below are arrays of numbers. 
 
 const sanDiegoSummerTemperatures = [79, 80, 72, 73, 82, 77, 76]
 // Expected output: [82, 80, 79, 77, 76, 73, 72]
+console.log(sanDiegoSummerTemperatures.sort((a, b) => (b - a))) //(a, b) is the compare function, comparing two numbers in the array
+//Output: [82, 80, 79, 77, 76, 73, 72]
 
 const sanDiegoWinterTemperatures = [59, 68, 62, 59, 66, 67, 66]
 // Expected output: [68, 67, 66, 66, 62, 59, 59]
+console.log(sanDiegoWinterTemperatures.sort((a, b) => (b - a))) //(a, b) is the compare function, comparing two numbers in the array and sorting based on whether they are negative, zero or positive repectfully
```
Comment (TsunamiCode5) [inline code-challenges.js:103]: This is a great practice! Leaving yourself notes on these assessments will get you in the habit of taking good documentation.

**Review by TsunamiCode5 — APPROVED** [review]
Mike, AWESOME job overall. You provided some really good pseudocode, your code looks really good, and your explanations in the coding concepts look great! As requested, feedback on your interview answers ... GREAT start. Really good job at putting it all in your own words and really thinking through your answers. I think that the examples you gave, definitions in your words, and the researched answers in your own words as well really drive it home that your understanding is solid. A way that you can really level up your interview questions, is to use them as a talking point! Elaborate even more if you can, provide a case scenario that it could be used in, or even take the opportunity to give an analogy. This assessment comprehension level is advanced beginner. Great work!

**Comment by mikerogers04** [comment]
Thank you for the wonderful feedback Nicole! I definitely can see how I can go even deeper into detail on interview answers with what you suggested. To clarify, I can merge and delete my branch now?

**Pull request merged**